### PR TITLE
Contextual Onboarding Linear

### PR DESCRIPTION
**Pull request by alessandroboron**
Task/Issue URL: https://app.asana.com/0/1206329551987282/1207812824986102/f

**Description**:

Show the Contextual linear flow. 

**Note**
I will add tests in a subsequent PR. I wanted to unblock you so you can use this branch.

**Steps to test this PR**:
1. Add temporary `newInstallCompletion(self)` in `DefaultVariantManager` at line 135
2. Enable environment variable VARIANT and set value to `mb`
3. Delete App from simulator.
4. Launch App.
5. Complete Onboarding flow

**Definition of Done (Internal Only)**:

* [ ] Does this PR satisfy our [Definition of Done](https://app.asana.com/0/1202500774821704/1207634633537039/f)?

**Copy Testing**:

* [ ] Use of correct apostrophes in new copy, ie `’` rather than `'`

**Orientation Testing**:

* [ ] Portrait
* [ ] Landscape

**Device Testing**:

* [ ] iPhone SE (1st Gen)
* [ ] iPhone 8
* [ ] iPhone X
* [ ] iPhone 14 Pro
* [ ] iPad

**OS Testing**:

* [ ] iOS 15
* [ ] iOS 16
* [ ] iOS 17

**Theme Testing**:

* [ ] Light theme
* [ ] Dark theme

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: DuckDuckGo/OnboardingExperiment/ContextualDaxDialogs/ContextualOnboardingDialogs.swift [hunk 1]
```diff
@@ -74,12 +75,13 @@ struct OnboardingFireButtonDialog: View {
...
 struct OnboardingFirstSearchDoneDialog: View {
-
-    @State var showNextScreen: Bool = false
-    @State var shouldFollowUp: Bool
-    let viewModel: OnboardingSiteSuggestionsViewModel
     let message = NSAttributedString(string: UserText.DaxOnboardingExperiment.ContextualOnboarding.onboardingFirstSearchDoneMessage)
     let cta = UserText.DaxOnboardingExperiment.ContextualOnboarding.onboardingGotItButton
+
+    @State private var showNextScreen: Bool = false
+    @State private var shouldFollowUp: Bool = true
```
Comment (SabrinaTardio) [inline DuckDuckGo/OnboardingExperiment/ContextualDaxDialogs/ContextualOnboardingDialogs.swift]: shouldn’t this come from outside? (based on wether or not a website has already been visited?

**Review by SabrinaTardio — COMMENTED** [review]
Linear Flow works as expected!
If you visit a site and then do a search, the search contextual appears and after pressing got it still shows the site suggestion.

I left a comment on this: showNextScreen in OnboardingFirstSearchDoneDialog should be set accordingly to whether a site has been visited or not

File: DuckDuckGo/DaxDialogs.swift [hunk 2]
```diff
@@ -317,7 +341,39 @@ final class DaxDialogs: NewTabDialogSpecProvider {
...
         return noTrackersMessage()
     }
-    
+
+    private func nextBrowsingMessageExperiment(privacyInfo: PrivacyInfo) -> BrowsingSpec? {
+        guard isEnabled, nextHomeScreenMessageOverride == nil else { return nil }
+        guard let host = privacyInfo.domain else { return nil }
+
+        if privacyInfo.url.isDuckDuckGoSearch {
+            // If user already performed a search, show after search dialog but don't follow up by suggesting to visit a site.
```
Comment (SabrinaTardio) [inline DuckDuckGo/DaxDialogs.swift]: I guess here should be “If user already visited a site..."

File: DuckDuckGoTests/TabViewControllerDaxDialogTests.swift [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+final class TabViewControllerDaxDialogTests: XCTestCase {
+    private var sut: TabViewController!
+    private var delegateMock: MockTabDelegate!
+    private var onboardingPresenterMock: ContextualOnboardingPresenterMock!
+
+    override func setUpWithError() throws {
+        try super.setUpWithError()
+        delegateMock = MockTabDelegate()
+        onboardingPresenterMock = ContextualOnboardingPresenterMock()
+        sut = .mock(contextualOnboardingPresenter: onboardingPresenterMock)
```
Comment (SabrinaTardio) [inline DuckDuckGoTests/TabViewControllerDaxDialogTests.swift]: I would not call this tab mock cause it’s a real tab I’d call it fixture? A standard initialisation

**Review by SabrinaTardio — APPROVED** [review]
Works great now! thanks!
I left 2 very minor comments

**Comment by SabrinaTardio** [comment]
> Works great now! thanks! I left 2 very minor comments

As discussed I merged https://github.com/duckduckgo/iOS/pull/3093 in here feel free to merge this when you have into the next branch etc 🥲

File: DuckDuckGoTests/TabViewControllerDaxDialogTests.swift (hunk 3 above)
Comment (alessandroboron) [reply DuckDuckGoTests/TabViewControllerDaxDialogTests.swift]: Changed to "fake"

**Comment by alessandroboron** [comment]
> Works great now! thanks! I left 2 very minor comments

Thanks, I’ve fixed them!

**Pull request merged**